### PR TITLE
MM-51792 Replace Boards feature flag with env var

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR replaces the Boards feature flag with an env var on the back-end.  A PR to remove from front-end is pending.

Currently Boards can be disabled via a feature flag that defaults to disabled. Multiple production incidents have happened on community due to not having the feature flag set.  The FF will be replaced with a env var that defaults to enabled.  

This is temporary until mm-server unit tests can run without disabling Boards.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51792

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/boards/server/boards_service_util.go [hunk 1]
```diff
@@ -80,12 +80,8 @@ func CreateBoardsConfig(mmconfig mm_model.Config, baseURL string, serverID strin
 		showFullName = *mmconfig.PrivacySettings.ShowFullName
 	}
 
-	serverRoot := baseURL + "/plugins/focalboard"
-	if mmconfig.FeatureFlags.BoardsProduct {
-		serverRoot = baseURL + "/boards"
-	}
 	return &config.Configuration{
-		ServerRoot:               serverRoot,
+		ServerRoot:               "/boards",
```
Comment (mgdelacroix) [inline server/boards/server/boards_service_util.go:84]:
```suggestion
		ServerRoot:               baseURL + "/boards",
```

**Review by lieut-data — APPROVED** [review]
Sweet, thanks @wiggin77!

**Comment by wiggin77** [comment]
> LGTM! +1 Only a small fix added

Actually that was a big fix. Thanks @mgdelacroix

**Comment by wiggin77** [comment]
I did a cloud server smoke test.  Looks good.

**Comment by lieut-data** [comment]
@wiggin77 & @amyblais, with our having already fast forwarded to v2023-04-13 for Cloud, should we plan to cherry-pick this for that release?

**Comment by amyblais** [comment]
@wiggin77 Feel free to cherry-pick to ``cloud`` branch.

**Pull request merged**